src/mono/mono/mini/monovm.c

```diff
@@ -181,6 +181,9 @@ parse_properties (int propertyCount, const char **propertyKeys, const char **pro
 		} else if (prop_len == 30 && !strncmp (propertyKeys [i], "System.Globalization.Invariant", 30)) {
 			// TODO: Ideally we should propagate this through AppContext options
 			g_setenv ("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", propertyValues [i], TRUE);
+		} else if (prop_len == 27 && !strncmp (propertyKeys [i], "System.Globalization.UseNls", 27)) {
```

Comment:
Does this really get passed from the default host? When was this added?

Reply:
Yeah. So these values are set by the user via `runtimeconfig.json` -- i.e:
https://github.com/dotnet/runtime/blob/e4b9fcf7da746cfe78e542e693683d15d3b9f75b/src/libraries/System.Runtime/tests/NlsTests/runtimeconfig.template.json#L1-L5

https://github.com/dotnet/runtime/blob/e4b9fcf7da746cfe78e542e693683d15d3b9f75b/src/libraries/System.Globalization/tests/Invariant/runtimeconfig.template.json#L1-L5

The host populates the `propertyKeys` and `propertyValues` sets with whatever is passed there and of course the `NATIVE_DLL_SEARCH_DIRECTORIES` and other initialization properties.

It seems like currently these properties are not propagated through `AppContext` in Mono so we set the fallback mono environment variable where Globalization reads this value from so that the user doesn't have to set those env vars. 

In coreclr, we get these values from AppContext first if we can't find it we fallback to env vars:
https://github.com/dotnet/runtime/blob/e4b9fcf7da746cfe78e542e693683d15d3b9f75b/src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs#L15-L28

Reply:
Yeah, I'm familiar with how the host/runtime interaction works (I wrote most of that Mono code). I'm wondering about that specific property, which I was not aware existed, but since it's in the tests I guess we're good. I guess it was only added a couple weeks ago?

Reply:
I see didn’t know that. Yeah the UseNls property support was introduced a couple of weeks ago for the ICU on Windows feature for costumers to disable using ICU and fallback to NLS.